pkg/deploy/generator/resources.go

```diff
@@ -44,6 +44,55 @@ func (g *generator) dnsZone(name string) *arm.Resource {
 	}
 }
 
+func (g *generator) networkSecurityPerimeter(name string) *arm.Resource {
+	return &arm.Resource{
+		Resource: &armnetwork.SecurityPerimeter{
+			Location:   pointerutils.ToPtr("[resourceGroup().location]"),
+			Properties: &armnetwork.SecurityPerimeterProperties{},
+			Name:       &name,
+			Type:       pointerutils.ToPtr("Microsoft.Network/networkSecurityPerimeters"),
+		},
+		APIVersion: azureclient.APIVersion("Microsoft.Network/networkSecurityPerimeters"),
+	}
+}
+
+// networkSecurityPerimeterProfile creates a new nsp profile with the hardcoded name `default`.
+func (g *generator) networkSecurityPerimeterProfile(nspName string) *arm.Resource {
+	return &arm.Resource{
+		Resource: &armnetwork.NspProfile{
+			Name: pointerutils.ToPtr(fmt.Sprintf("%s/default", nspName)),
+			Type: pointerutils.ToPtr("Microsoft.Network/networkSecurityPerimeters/profiles"),
+		},
+		APIVersion: azureclient.APIVersion("Microsoft.Network/networkSecurityPerimeters/profiles"),
+		DependsOn: []string{
+			fmt.Sprintf("[resourceId('Microsoft.Network/networkSecurityPerimeters', '%s')]", nspName),
+		},
+	}
+}
+
+func (g *generator) networkSecurityPerimeterAssociation(nspName string, associationName string, targetResourceId string) *arm.Resource {
+	return &arm.Resource{
+		Resource: &armnetwork.NspAssociation{
+			Properties: &armnetwork.NspAssociationProperties{
+				AccessMode: pointerutils.ToPtr(armnetwork.AssociationAccessModeLearning),
```

Comment:
minor: if some day we want to move this to enforce or audit... would it be worth to have it as a func parameter? I like how clean looks right now, so it's not a deal breaker.